xonsh/procs/posix.py

```diff
@@ -52,6 +52,9 @@ class PopenThread(threading.Thread):
 
     def __init__(self, *args, stdin=None, stdout=None, stderr=None, **kwargs):
         super().__init__()
+
+        self.daemon = True
+
```

Comment:
I haven't dived deep into the proc handling in a bit -- was the absence of this causing a breakage somewhere?

Reply:
This option is required to support `disown` and `nohup` because otherwise the shell exit will block waiting for subprocess completion.

Outside of that case, I believe this is the proper behavior since it will cause the shell to orphan frozen processes (ie, subprocess is stuck in a syscall) instead of waiting for them.